assembly-ide/src/assembly/bin/che.sh

```diff
@@ -10,20 +10,436 @@
+  export CATALINA_HOME="${CHE_HOME}/tomcat"
+
+  # Convert windows path name to POSIX
+  if [[ $CATALINA_HOME == *":"* ]]
+  then 
+    CATALINA_HOME=$(echo /$CATALINA_HOME | sed  's|\\|/|g' | sed 's|:||g')
+  fi
+
+  export CATALINA_BASE="${CHE_HOME}/tomcat"
+  export ASSEMBLY_BIN_DIR="${CATALINA_HOME}/bin"
+
+  # Global logs directory
+  if [ -z "${CHE_LOGS_DIR}" ] || [ ! -f "${CHE_LOGS_DIR}" ]; then
+    export CHE_LOGS_DIR="${CATALINA_HOME}/logs/"
+  fi
+}
+
+function get_docker_ready {
+  # Create absolute file names for docker and docker-machine
+  # DOCKER_TOOLBOX_INSTALL_PATH set globally by Docker Toolbox installer
+  if $win ; then
+    if [ ! -z "$DOCKER_TOOLBOX_INSTALL_PATH" ]; then
+      export DOCKER_MACHINE=${DOCKER_TOOLBOX_INSTALL_PATH}\\docker-machine.exe
+      export DOCKER=${DOCKER_TOOLBOX_INSTALL_PATH}\\docker.exe
+    else
+      error_exit "!!! DOCKER_TOOL_BOX_INSTALL_PATH environment variable not set. Add it or rerun Docker Toolbox installation.!!!"
+      return
+    fi
+  elif $mac ; then
+    export DOCKER_MACHINE=/usr/local/bin/docker-machine
+    export DOCKER=/usr/local/bin/docker
+  elif $linux ; then
+    export DOCKER_MACHINE=
+    export DOCKER=/usr/bin/docker
+  fi 
+
+  ### If Windows or Mac, launch docker-machine, if necessary
+  if $win || $mac ; then
+    # Path to run VirtualBox on the command line - used for creating VMs
+    if [ ! -z "$VBOX_MSI_INSTALL_PATH" ]; then
+      VBOXMANAGE=${VBOX_MSI_INSTALL_PATH}VBoxManage.exe
+    else 
+      VBOXMANAGE=/usr/local/bin/VBoxManage
+    fi
+
+    if [ ! -f "${DOCKER_MACHINE}" ]; then
+      error_exit "!!! Could not find docker-machine executable. Win: DOCKER_TOOL_BOX_INSTALL_PATH env variable not set. Add it or rerun Docker Toolbox installation. Mac: Expected docker-machine at /usr/local/bin/docker-machine. !!!"
+      return
+    fi
+
+    if [ ! -f "${VBOXMANAGE}" ]; then
+      error_exit "!!! Could not find VirtualBox. Win: VBOX_MSI_INSTALL_PATH env variable not set. Add it or rerun Docker Toolbox installation. Mac: Expected Virtual Box at /usr/local/bin/VBoxManage. !!!"
+      return
+    fi
+
+    # Test to see if the VM we need is already running
+    "${VBOXMANAGE}" showvminfo $VM &> /dev/null
+    VM_EXISTS_CODE=$?
+
+    # Stops execution of script if there is an error
+    #set -e
+
+    if [ $VM_EXISTS_CODE -eq 1 ]; then
+      echo "Creating docker machine named $VM..."
+      "$DOCKER_MACHINE" rm -f $VM &> /dev/null || :
+      rm -rf ~/.docker/machine/machines/$VM
+      "$DOCKER_MACHINE" create -d virtualbox $VM
+    else
+      echo -e "Docker machine named ${GREEN}$VM${NC} already exists..."
+    fi
+
+    VM_STATUS=$("${DOCKER_MACHINE}" status $VM 2>&1)
+
+    if [ "$VM_STATUS" != "Running" ]; then
+      echo "Starting machine named $VM..."
+      "${DOCKER_MACHINE}" start $VM
+      yes | "${DOCKER_MACHINE}" regenerate-certs $VM
+    fi
+
+    echo -e "Setting environment variables for machine ${GREEN}$VM${NC}..."
```

Comment:
yes, the ${GREEN} and ${NC} constants are not processed properly on windows if you do not include the -e option.